packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -172,11 +190,11 @@ export class MemoryStore {
       });
     }
 
-    if (protocols) {
-      protocols.forEach(protocol => {
-        if (!this._protocols.find(p => _.isEqual(p, protocol))) {
-          this._protocols.push(protocol);
-          this._eventEmitter.emit('newProtocol', protocol);
+    if (objectives) {
+      objectives.forEach(objective => {
```

Comment:
```suggestion
      objectives?.forEach(objective => {
```

Using [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining), we can get rid of the `if (objectives) { // type is narrowed }`